### ISSUE TITLE
Make the AllowInnerSlashes option for Style/RegexpLiteral more configurable

### ISSUE THREAD

**Issue by ChallahuAkbar** [bbatsov/rubocop#3237]
I did a quick search of the issues to see if anyone else had asked about this and didn't see it so sorry if this as already been answered/is in the wrong place.
I've recently started working on a fairly large existing codebase, and the styleguide specified is [GitHub's](https://github.com/styleguide/ruby). While GitHub's and yours are fairly similar, there are some differences. One such difference is using the %r literal when there are backslashes in the regular expression. While you recommend using it when one or more slashes appear in the regex, GitHub recommends using it only when _more_ than one backslash appears.

Point is, I'd like to keep this cop enabled (and not have to disable it just for the lines that don't work), but because the AllowInnerSlashes option hits when there's one slash in the regex, it ends up throwing a number of false positives for my use case.

(as a sidenote, I suppose another solution to this would be to try and convince the rest of the team to ignore this particular bit of GitHub styleguide silliness...)

--------
## RuboCop version

Include the output of `rubocop -V`:

```
$ rubocop -V
0.40.0 (using Parser 2.3.1.2, running on ruby 2.3.0 x86_64-linux-gnu)
```

**Comment by jonas054** [comment]
You'll be interested to learn that we used to have a configuration option for this, but it was removed in #1721. You can follow the references from there. The key thing is that @bbatsov said in #1655 that

> I think MaxSlashes was a mistake - you either allow them or don't.

**Comment by ChallahuAkbar** [comment]
Ah, guess my search wasn't thorough enough. That's unfortunate; I understand the reasoning but it's too bad that there isn't really a good way around this. Guess I'll talk to the team and figure out which way we wanna go. Thanks for the quick response!
